apps/api/src/app/events/usecases/trigger-event/trigger-event.usecase.ts

```diff
@@ -1,65 +1,109 @@
 import { JobEntity, JobRepository, NotificationTemplateRepository, NotificationRepository } from '@novu/dal';
-import { Injectable } from '@nestjs/common';
+import { Injectable, Logger } from '@nestjs/common';
 import { StepTypeEnum, ExecutionDetailsSourceEnum, ExecutionDetailsStatusEnum } from '@novu/shared';
 import * as Sentry from '@sentry/node';
 
 import { TriggerEventCommand } from './trigger-event.command';
-import { ProcessSubscriber } from '../process-subscriber/process-subscriber.usecase';
-import { ProcessSubscriberCommand } from '../process-subscriber/process-subscriber.command';
-import { ApiException } from '../../../shared/exceptions/api.exception';
+
 import { AddJob } from '../add-job/add-job.usecase';
+import { CreateNotificationCommand, CreateNotification } from '../create-notification';
+import { ProcessSubscriber, ProcessSubscriberCommand } from '../process-subscriber';
+
 import { CreateExecutionDetails } from '../../../execution-details/usecases/create-execution-details/create-execution-details.usecase';
 import {
   CreateExecutionDetailsCommand,
   DetailEnum,
 } from '../../../execution-details/usecases/create-execution-details/create-execution-details.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { Cached } from '../../../shared/interceptors';
+import { CacheKeyPrefixEnum } from '../../../shared/services/cache';
+
+const LOG_CONTEXT = 'TriggerEventUseCase';
 
 @Injectable()
 export class TriggerEvent {
   constructor(
-    private notificationTemplateRepository: NotificationTemplateRepository,
+    private addJobUsecase: AddJob,
+    private createNotification: CreateNotification,
     private processSubscriber: ProcessSubscriber,
     private jobRepository: JobRepository,
-    private addJobUsecase: AddJob,
     private notificationRepository: NotificationRepository,
+    private notificationTemplateRepository: NotificationTemplateRepository,
     protected createExecutionDetails: CreateExecutionDetails
   ) {}
 
   async execute(command: TriggerEventCommand) {
-    await this.validateTransactionIdProperty(command.transactionId, command.organizationId, command.environmentId);
+    const { actor, environmentId, identifier, organizationId, to, userId } = command;
 
-    const template = await this.notificationTemplateRepository.findByTriggerIdentifier(
-      command.environmentId,
-      command.identifier
-    );
+    await this.validateTransactionIdProperty(command.transactionId, organizationId, environmentId);
 
     Sentry.addBreadcrumb({
       message: 'Sending trigger',
       data: {
-        triggerIdentifier: command.identifier,
+        triggerIdentifier: identifier,
       },
     });
 
+    const template = await this.notificationTemplateRepository.findByTriggerIdentifier(
+      command.environmentId,
+      command.identifier
+    );
```

Comment:
Potentially we could send the template from the queued data as during ParseEventRequest we already check the template exists, but I find sensible to retrieve it again in case during the queued time for the event, the template is being somehow deleted.